src/qt/importdialog.cpp

```diff
@@ -298,5 +344,377 @@ void ImportDialog::on_Accept_clicked() {
+                    nLowestTimestamp = timestamp;
+                }
+            }
+
+            if (fRescan && fRunScan && ui->entries->count()) {
+                int64_t scannedTime = walletModel->wallet().RescanFromTime(nLowestTimestamp, reserver, true);
+                walletModel->wallet().ReacceptWalletTransactions();
+
+                if (walletModel->wallet().IsAbortingRescan()) {
+                    QMessageBox::critical(this, tr("Misc Error"),
+                                          tr("Rescan aborted by user."));
+
+                    if (scannedTime > nLowestTimestamp) {
+                        std::vector <resultpdi> results = response;
+                        response.clear();
+
+                        for (int i = 0; i < ui->entries->count(); ++i) {
+                            ImportEntry *entry = qobject_cast<ImportEntry *>(ui->entries->itemAt(i)->widget());
+                            // If key creation date is within the successfully scanned
+                            // range, or if the import result already has an error set, let
+                            // the result stand unmodified. Otherwise, replace the result
+                            // with an error message.
+                            if (scannedTime <= GetImportTimestamp(entry->getMultiData().timestamp, now) || !results.at(i).error.empty()) {
+                                response.push_back(results.at(i));
+                            } else {
+                                resultpdi result;
+                                result.success = false;
+                                result.error = strprintf(
+                                                "Rescan failed for key with creation timestamp %d. There was an error reading a "
+                                                "block from time %d, which is after or within %d seconds of key creation, and "
+                                                "could contain transactions pertaining to the key. As a result, transactions "
+                                                "and coins using this key may not appear in the wallet. This error could be "
+                                                "caused by pruning or data corruption (see bitcoind log for details) and could "
+                                                "be dealt with by downloading and rescanning the relevant blocks (see -reindex "
+                                                "option and rescanblockchain RPC).",
+                                                GetImportTimestamp(entry->getMultiData().timestamp, now), scannedTime - TIMESTAMP_WINDOW - 1,
+                                                TIMESTAMP_WINDOW);
+                                response.push_back(std::move(result));
+                            }
+                        }
+                    }
+                }
+            }
+            for (int i = 0; i < response.size(); i++) {
```

Comment:
In c71f418f18000f537871f57a42ec76e0a25941e9 "qt: implemented importmulti and importdescriptors GUI's"

Compiler warning here, also below for `importDescriptors`:

```
qt/importdialog.cpp: In member function ‘void ImportDialog::on_Accept_clicked()’:
qt/importdialog.cpp:496:31: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<resultpdi>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  496 |             for (int i = 0; i < response.size(); i++) {
      |                             ~~^~~~~~~~~~~~~~~~~
```

```suggestion
            for (size_t i = 0; i < response.size(); i++) {
```

Reply:
done